### PR TITLE
Data contribution 2

### PR DESCRIPTION
**Pull request by nafeldman**
Hi Education Team! Please review as your time allows. Our goal is to have these ready to be published by the end of the month.

### CODE REVIEW

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:12]:
```suggestion
comment: Learn about how to disclose data to CHOP by making a data contribution!
long_description:  Arcus Archives serves as the canonical repository for research data at CHOP. Archiving data with Arcus safeguards and maintains research data in accordance with established archival protocols, also while promoting data sharing. The Data Contribution Module provides an overview of the reasons to archive research data with Arcus, as well as the processes involved in scoping data, conducting privacy assessments, and addressing technical requirements prior to data submission. This Module explores the Project Template directory structure, and delves into the specific steps required for archiving data within the Arcus platform.
```

Note that you're missing a "long_description" so I've taken the liberty of adding a comment and making your comment the long_description.  Feel free to tweak, this is just my suggestion.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:110]:
```suggestion
If after viewing this module, you are prepared to archive data with Arcus, please fill out [a data contribution request](https://pm.arcus.chop.edu/servicedesk/customer/portal/6/create/256) to start the process.
```
When I saw the word "following" I was looking for another line...

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:124]:
```suggestion
Arcus's goal for research data management and the project template is to provide tools that are relevant throughout the entire lifecycle of research data. The project template is designed to be adaptable and iterative to capture the wide range of research activities at CHOP. The project template combines flexibility to encompass diverse data capturing needs, while also maintaining a consist structure for all archived data which facilitates seamless communication among various projects spanning different domains, thereby promoting effective data sharing.
```
I *think* the current "possessive apostrophe following S" rule includes another s?  But feel free to disregard, I haven't actually researched this.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:143]: Ideally any text on this image (and other images that similarly convey information via text) is captured in alt text or other text elsewhere in the document.  This is important for accessibility, mobile viewing where an image might be v small on a phone, as well as search.  But don't let this hold you back, we can't let the perfect be the enemy of the good!

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:166]: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases.  I leave it to you to decide if the effort of re-doing is worth it.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:149]: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:172]: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:178]: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:184]: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:141]: These links don't actually take you into the applications, so consider something like:

```suggestion
The Project Template prioritizes streamlined archiving and reproducible research pathways. It archives a wide range of research types from the Research Institute, making them discoverable through tools like Arcus Cohort Discovery, available as an application [on the Arcus website](https://arcus.chop.edu), [Gene](https://chop.alationcloud.com), and Omics Variant Browser, [available as an application [on the Arcus website](https://arcus.chop.edu). The Project Template facilitates organizing diverse research data in a single directory structure, enabling automated archiving, metadata management, and data delivery throughout the research data lifecycle. This file directory structure is used for the entire lifecycle of research data within Arcus:
```

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:235]: Clicking these, I'm left with a big empty space afterwards that looks like nothing happened unless I scroll down the page.  Consider checking the functions?  I think there may be a missing image or one that is being hidden?

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:245]: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:266]: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:324]: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:341]: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:372]: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:392]: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:465]: Similarly, relying on an image here without copying the text into alt text or supplemental text may hinder usability in some cases. I leave it to you to decide if the effort of re-doing is worth it.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:474]:
```suggestion
The file\_manifest.csv matches the biosample\_id to each file in the data folders. Below is more detail about each section in the file:
```
If you don't escape underscores, markdown thinks you want the enclosed text italicized if there are two underscores on the same line.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:476]:
```suggestion
- biosample\_id is an ID number for each file. For some studies, each file is derived about specific biosamples, so we suggest using the sample id. ideally, biosample\_id links to the CHOP biobank. When you cannot link the the biobank, treat biosample\_id as the IDs you use for samples taken from participants. For studies where there are no biosamples, the biosample\_id can be the file name.
```

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:479]:
```suggestion
- file\_path is the file path for each file in the **\_data\_** folders. File paths should start with **\_data\_** and end with the full file name with extension
```

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:494]:
```suggestion
For Omics data, the **file\_manifest.csv** matches biosample IDs to data files and experimental protocols, described in \_yaml\_ files. More information about the protocols files is available in the references section of this module. Many files might share the same experimental protocol. These yaml protocol files describe experiment and data processing details. See below for the columns in a file\_manifest:
```

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:503]:
```suggestion
| derived\_from\_file\_group | This column describes relations between file groups. We want to capture consecutive pipeline steps. For example, a bam file is derived from a paired fastq group. Use the name of the file_groups used to construct this file. Delimit multiple groups with a semicolon. Use NA when there are no prior step files to reference. | String |
```
Strangely this is still showing up with weird italicizing for me, even though you're escaping.  Not sure what's up?

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:536]:
```suggestion
- local\_participant\_id is a local identifier the study team used to identify the patient
```

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:553]: This is still showing up weirdly italicized for me?  Again, not sure what's up

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:557]: This is still showing up weirdly italicized for me? Again, not sure what's up

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:561]: This is still showing up weirdly italicized for me? Again, not sure what's up

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:622]:
```suggestion

<!-- data-type="none" -->
```
Do you want the graph type "parallel crosswalk" to appear?  If not, I suggest you add the data type to the top of the table.

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:656]:
```suggestion
| destination\_file\_group | The files in this file group is derived from source\_file\_group. | String |
```

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:657]:
```suggestion
| source\_file\_group      | File group used to derive the destination\_file\_group.           | String |
```

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:661]:
```suggestion
For each script/notebook in \_src/\_, and each model in \_models/\_, there should be an env file (here env refers to a file named env with any extension, so env.yaml or env.txt, for example) that describes the environment in which it was created or run. Environment files should be named as follows: **descriptiveName\_env** and placed in a folder called environments within the \_configs/environment\_ directory. Either individuals files or entire folders (whichever is the appropriate level) in scripts and notebooks within the \_src/\_ directory, or the \_models/\_ directory will need to be added to the env manifest file, matching them with their related environment file. See the see below for more information about this file:
``` 

Again, I think I'm interpreting what you're doing... that you want these underscores to appear, right?  If you have two or more in a single line of text, you **must** prefix them with the backslash to escape their normal markdown usage (italics)

File: data_contribution_2_data.md
Comment (pm0kjp) [inline data_contribution_2_data.md:671]: Link doesn't go where you want it to, I think?

**Review by pm0kjp — CHANGES_REQUESTED** [review]
This looks good but I realized at around line 600 that I keep repeating the same things so I wanted to stop to give you a chance to look at what I've commented so far before I keep copy/pasting.  The basic things I keep repeating:

* Using text on an image to drive understanding is suboptimal for various reasons, but I can imagine that adding text into the document itself might be a huge PITA and worth doing as a "secondary improvement", so I won't hinder you moving forward on this
* There are lots of unintentional italics where you have 2 or more underscores on the same line of text without escaping them.

Take a look at what I've suggested (and continue looking especially for unintentional italics further on, I stopped noting them but you can see your script *in situ* at https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/Arcus_Labs_Orientation/data-contribution-2/data_contribution_2_data.md#1 to see what it looks like)

**Comment by nafeldman** [comment]
thank you @pm0kjp - that's a really good point about adding alt text (especially to the gifs) we should definitely go back and do that.

File: data_contribution_2_data.md
Comment (allrolsen) [reply data_contribution_2_data.md]: Thank you for pointing this out Joy! We reused graphics we already has for this module, which is why there are so many images with text. We are not going to be able to lay these images out in CSS to match W3 standards. I went through and at least improved all the alt text. We will keep in mind these alt text standards in future.

File: data_contribution_2_data.md
Comment (allrolsen) [reply data_contribution_2_data.md:662]: Thank you for catching these. Nicole and I thought we had corrected all of them, but more keep on showing up!

File: data_contribution_2_data.md
Comment (allrolsen) [reply data_contribution_2_data.md:623]: We did not want a 'parallel crosswalk' and did not know how to get rid of it!

File: data_contribution_2_data.md
Comment (allrolsen) [reply data_contribution_2_data.md]: Thank you, I removed the link

**Comment by allrolsen** [comment]
> This looks good but I realized at around line 600 that I keep repeating the same things so I wanted to stop to give you a chance to look at what I've commented so far before I keep copy/pasting. The basic things I keep repeating:
> 
> * Using text on an image to drive understanding is suboptimal for various reasons, but I can imagine that adding text into the document itself might be a huge PITA and worth doing as a "secondary improvement", so I won't hinder you moving forward on this
> * There are lots of unintentional italics where you have 2 or more underscores on the same line of text without escaping them.
> 
> Take a look at what I've suggested (and continue looking especially for unintentional italics further on, I stopped noting them but you can see your script _in situ_ at https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/Arcus_Labs_Orientation/data-contribution-2/data_contribution_2_data.md#1 to see what it looks like)

@pm0kjp Thank you for all of your edits and suggestions. I have gone through the text and ensured that all of the _ are properly escape. Despite this, some text is still italicizing. All the text that still does this is with the scripts. Is there someone we can reach out to at LiaScript to look at this?

**Comment by rosemm** [comment]
Hi @allrolsen I noticed the problem you pointed out about the escaped underscores not working within the contingent text. It's because that text is piped in as part of a script -- anything that needs to be escaped in that text has to be escaped twice (e.g. `\\_`) because one "layer" of the escaping happens as the text is parsed by the script, and you still need an escape present there when it comes out the other side and gets inserted into the page. 
I went through and put those extra escapes in for you, and I think it's all rendering correctly now --- let me know if you still see problems!

**Comment by rosemm** [comment]
I also saw a few typos. Do you have spell check enabled in your text editor? (https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) I'm a terrible speller, so I rely on it a lot :) It saves me a ton of time and errors.

**Comment by allrolsen** [comment]
> Hi @allrolsen I noticed the problem you pointed out about the escaped underscores not working within the contingent text. It's because that text is piped in as part of a script -- anything that needs to be escaped in that text has to be escaped twice (e.g. `\\_`) because one "layer" of the escaping happens as the text is parsed by the script, and you still need an escape present there when it comes out the other side and gets inserted into the page.
> I went through and put those extra escapes in for you, and I think it's all rendering correctly now --- let me know if you still see problems!

Thank you!! Nicole and I were really annoyed by this, and had no idea how to fix it.

**Comment by allrolsen** [comment]
> I also saw a few typos. Do you have spell check enabled in your text editor? (https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) I'm a terrible speller, so I rely on it a lot :) It saves me a ton of time and errors.

I did not have spell check enabled, thank you for sharing that tool. I added it to VS Code and fixed all the spelling mistakes.

**Comment by allrolsen** [comment]
@pm0kjp and @rosemm Thank you both again for reviewing this module, your suggestions and help are invaluable! @nafeldman and I fell it looks good and are ready to merge this branch. I am not able to do it, it says there are changes requested (believe I resolved all of them) and that there is a conflict (media/source/.DS_Store file). Could one of you merge it? Someone with an Admin role should be over to override these issues.

**Review by rosemm — APPROVED** [review]

**Comment by rosemm** [comment]
I fixed the merge conflict, so it should be ready to go now. 
Unfortunately, it looks like github won't let me merge it without @pm0kjp specifically signing off, since she had requested changes earlier. Joy's out today, so it might be a little while before she has a chance to approve.

**Review by pm0kjp — APPROVED** [review]

**Pull request merged**
